ngx_pagespeed can't be built with clang/clang++

`CC=clang CXX=clang++ ./configure --add-module=/tmp/gnx/ngx_pagespeed`

```
clang -c -pipe  -O -Wall -Wextra -Wpointer-arith -Wconditional-uninitialized -Wno-unused-parameter -Werror -g    -I src/core -I src/event -I src/event/modules -I src/os/unix -I /tmp/gnx/ngx_pagespeed/psol/include -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/chromium/src -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/google-sparsehash/src -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/google-sparsehash/gen/arch/linux/x64/include -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/protobuf/src -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/re2/src -I /tmp/gnx/ngx_pagespeed/psol/include/out/Release/obj/gen -I /tmp/gnx/ngx_pagespeed/psol/include/out/Release/obj/gen/protoc_out/instaweb -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/apr/src/include -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/aprutil/src/include -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/apr/gen/arch/linux/x64/include -I /tmp/gnx/ngx_pagespeed/psol/include/third_party/aprutil/gen/arch/linux/x64/include -I objs -I src/http -I src/http/modules -I src/mail \
                -o objs/addon/src/ngx_pagespeed.o \
                /tmp/gnx/ngx_pagespeed/src/ngx_pagespeed.cc
In file included from /tmp/gnx/ngx_pagespeed/src/ngx_pagespeed.cc:36:
In file included from /tmp/gnx/ngx_pagespeed/src/ngx_rewrite_driver_factory.h:31:
In file included from /tmp/gnx/ngx_pagespeed/psol/include/net/instaweb/system/public/system_rewrite_driver_factory.h:25:
/tmp/gnx/ngx_pagespeed/psol/include/net/instaweb/rewriter/public/rewrite_options.h:56:1: error: class 'HttpOptions' was previously declared as a struct [-Werror,-Wmismatched-tags]
class HttpOptions;
^
/tmp/gnx/ngx_pagespeed/psol/include/pagespeed/kernel/http/http_options.h:25:8: note: previous use is here
struct HttpOptions {
       ^
/tmp/gnx/ngx_pagespeed/psol/include/net/instaweb/rewriter/public/rewrite_options.h:56:1: note: did you mean struct here?
class HttpOptions;
^~~~~
struct
In file included from /tmp/gnx/ngx_pagespeed/src/ngx_pagespeed.cc:50:
In file included from /tmp/gnx/ngx_pagespeed/psol/include/net/instaweb/rewriter/public/resource_fetch.h:29:
In file included from /tmp/gnx/ngx_pagespeed/psol/include/net/instaweb/util/public/google_url.h:20:
In file included from /tmp/gnx/ngx_pagespeed/psol/include/pagespeed/kernel/http/google_url.h:31:
In file included from /tmp/gnx/ngx_pagespeed/psol/include/third_party/chromium/src/url/gurl.h:14:
/tmp/gnx/ngx_pagespeed/psol/include/third_party/chromium/src/url/url_canon_stdstring.h:44:31: error: 'override' keyword is a C++11 extension [-Werror,-Wc++11-extensions]
  virtual void Resize(int sz) OVERRIDE;
                              ^
/tmp/gnx/ngx_pagespeed/psol/include/third_party/chromium/src/base/compiler_specific.h:143:18: note: expanded from macro 'OVERRIDE'
#define OVERRIDE override
                 ^
/tmp/gnx/ngx_pagespeed/src/ngx_pagespeed.cc:2307:43: error: variable 'inflate_type' may be uninitialized when used here [-Werror,-Wconditional-uninitialized]
        ctx->inflater_ = new GzipInflater(inflate_type);
                                          ^~~~~~~~~~~~
/tmp/gnx/ngx_pagespeed/src/ngx_pagespeed.cc:2294:7: note: variable 'inflate_type' is declared here
      GzipInflater::InflateType inflate_type;
      ^
3 errors generated.
make[1]: *** [objs/addon/src/ngx_pagespeed.o] Error 1
make[1]: Leaving directory `/tmp/gnx/nginx-1.6.2'
make: *** [build] Error 2

```

--- a/src/ngx_pagespeed.cc
+++ b/src/ngx_pagespeed.cc
@@ -2291,14 +2291,13 @@ ngx_int_t ps_html_rewrite_header_filter(ngx_http_request_t* r) {
     if (!ps_has_stacked_content_encoding(r)) {
       StringPiece content_encoding =
           str_to_string_piece(r->headers_out.content_encoding->value);
-      GzipInflater::InflateType inflate_type;
+      GzipInflater::InflateType inflate_type = GzipInflater::kGzip;
       bool is_encoded = false;
       if (StringCaseEqual(content_encoding, "deflate")) {
         is_encoded = true;
         inflate_type = GzipInflater::kDeflate;
       } else if (StringCaseEqual(content_encoding, "gzip")) {
         is_encoded = true;
-        inflate_type = GzipInflater::kGzip;
       }
 
       if (is_encoded) {

--- a/src/ngx_server_context.cc
+++ b/src/ngx_server_context.cc
@@ -48,7 +48,7 @@ SystemRequestContext* NgxServerContext::NewRequestContext(
     ngx_http_request_t* r) {
   // Based on ngx_http_variable_server_port.
   bool port_set = false;
-  int local_port;
+  int local_port = 0;
 #if (NGX_HAVE_INET6)
   if (r->connection->local_sockaddr->sa_family == AF_INET6) {
     local_port = ntohs(reinterpret_cast<struct sockaddr_in6*>(

